justingolden21/desktopclock
# Fix mobile nav when too many icons pinned

### Problem:

On mobile, when users have the "smaller menu" option checked (Settings > Appearance > Smaller menu)

![image](https://github.com/justingolden21/desktopclock/assets/30274440/99d7366b-06a0-478f-a3bd-65160333cd8f)

If they have many items pinned and a small height screen, the icons become very small:

![image](https://github.com/justingolden21/desktopclock/assets/30274440/9b928f92-1744-4619-8a30-98f50f4fefee)

At a slightly larger screen, it looks correct:

![image](https://github.com/justingolden21/desktopclock/assets/30274440/1ba3057b-d383-4bfb-aa30-c29f1f5250df)

This works fine with the normal menu:

![image](https://github.com/justingolden21/desktopclock/assets/30274440/8a2d9b9d-d941-49e9-8176-afb39fa9d50e)

### Solution

The reason for the problem is the width and max-width we set in the menu. These are actually not necessary at all, since the width can be determined by the icons themselves, which are always the same size. The reason we set the max-width for the normal menu is in case some items (for example, "Rubiks Cube Timer") have very long names and we want it to wrap on a new line, and want the navbar to be a consistent width regardless of language and pinned items (for design, user consistency, and to better design UI around this).

### Testing Notes

1. Check Settings > Appearance > Smaller menu
2. Pin every app
3. Resize the screen to a small height mobile device
4. Confirm that the icons are normally sized and overflow correctly:

![image](https://github.com/justingolden21/desktopclock/assets/30274440/b25813ac-de33-4ac7-acc7-d1c0e731cc43)

Confirm that the nav bar works on all screen sizes and for all settings (test mobile, desktop, and fullscreen, and each combination of smaller menu on/off and always collapse menu on/off) with and without overflow.



https://github.com/justingolden21/desktopclock/assets/30274440/aa3969e9-d26c-478d-a5cf-7c1f83390b24
